src/lib/CouchDB.ts

```diff
@@ -82,46 +98,67 @@ class CouchDB {
     return authHeader;
   }
 
-  async #createDatabase() {
-    Log(
-      { msg: "Creating Database if not exists.", dbUrl: this.#dbUrl },
+  async createDatabase(options?: dbUrlOptsIF) {
+    const { dbUrl, authHeader } = this.#getDbParams(options);
```

Comment:
Added options parameters to all the CouchDB methods to allow overriding any option at call time